### ISSUE TITLE
GroupMeClientApi.Groups() is returning empty.

### ISSUE THREAD

**Issue by minecraftchest1** [alexdillon/GroupMeClientApi#30]
`GroupMeClientApi.Groups()` is returning an empty collection.

## Program Output 
```
1: Login/Change User
2: List groups
3: List Direct Messages
4: Who am I
0: Exit
Please enter a menu item number: 2
Chats
Token: ********************

~/src/groupmeCLI$
```

## Code that gives me the problem.
```
static async Task _groups()
{
	Console.WriteLine("Chats");
	string _token = _getToken();
	Console.WriteLine("Token: ********************", _token);
	Console.WriteLine();
	GroupMeClientApi.GroupMeClient _client = new GroupMeClientApi.GroupMeClient(_token);
		await _client.GetGroupsAsync();
	await _client.GetChatsAsync();
	await _client.GetContactsAsync();
		foreach (IMessageContainer _messageContainer in _client.Chats())
	{
		Console.WriteLine("Chat name: {0}", _messageContainer.Name);
	}
}
```
Full code at https://github.com/minecraftchest1/GroupMeCLI

## More information.
I have tested accessing the API fr groups from firefox and it returns several groups.

**Comment by alexdillon** [comment]
Hi, thanks for checking out my project!

I looked at the code snippet you provided, as well as the repo you linked to for your GroupMe CLI. I believe this may be a threading issue that you're seeing. When I ran your project, I was able to replicate the behavior you saw, with the program terminating after running `await _client.GetGroupsAsync()` (line 99 of program.cs).

Changing your call to the groups function on line 40 to `_groups().Wait();` appears to solve the issue, and your code runs as expected. Alternatively, it worked when I changed your main method to be `static async Task Main()`, and changed line 40 to be `await _groups();`.

Let me know if this doesn't help and I'll be glad to look into it more.
